core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -69,7 +71,7 @@
      * @param endOfStream whether the frame is the last one for the GRPC stream
      * @param flush {@code true} if more data may not be arriving soon
      */
-    void deliverFrame(WritableBuffer frame, boolean endOfStream, boolean flush);
+    void deliverFrame(@Nullable WritableBuffer frame, boolean endOfStream, boolean flush);
```

Comment:
It's a pre-existing optimization. I just put in `@Nullable` to make it more clear. It is used instead of a zero-length `frame` when needing to signal the transport to `flush`.